torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -6038,37 +6042,70 @@ def test_cuda_future_device_not_cuda(self):
         ):
             fut = Future(devices=["cpu"])
 
-    def _test_cuda_future_extraction(self, wrapper, unwrapper):
-        # We check proper CUDA stream synchronization by filling the tensor with
-        # the expected value in one stream, and reading it from another stream.
-        tensor = torch.zeros((100,), device="cuda:0")
+    def _test_cuda_future_extraction(self, wrapper, unwrapper, sparse_tensor):
+        # We check proper CUDA stream synchronization by adding to the tensor
+        # in one stream to get the expected value, and reading it from another stream.
+        if sparse_tensor:
+            tensor = build_sparse_tensor().to("cuda:0")
+            add_tensor = build_sparse_tensor().to("cuda:0")
+            expected_tensor = (tensor + add_tensor).coalesce()
+        else:
+            tensor = torch.zeros((100,), device="cuda:0")
+            add_tensor = torch.ones((100,), device="cuda:0")
+            expected_tensor = tensor + add_tensor
         future = Future(devices=["cuda:0"])
         with torch.cuda.device("cuda:0"):
             stream = torch.cuda.Stream()
             another_stream = torch.cuda.Stream()
             with torch.cuda.stream(stream):
                 torch.cuda._sleep(int(1000 * get_cycles_per_ms()))
-                tensor.fill_(1)
+                tensor += add_tensor
+                if sparse_tensor:
+                    tensor = tensor.coalesce()
                 future.set_result(wrapper(tensor))
             with torch.cuda.stream(another_stream):
-                self.assertTrue(torch.eq(unwrapper(future.wait()), 1).all().item())
+                tensor = unwrapper(future.wait())
+                if sparse_tensor:
+                    self.assertTrue(torch.eq(tensor.indices(), expected_tensor.indices()).all().item())
+                    self.assertTrue(torch.eq(tensor.values(), expected_tensor.values()).all().item())
+                    self.assertTrue(torch.eq(torch.tensor(tensor.size()), torch.tensor(expected_tensor.size())).all().item())
```

Comment:
Is there a reason for comparing the sizes by first converting them to a tensor and then doing tensor comparison? I think sizes are tuples hence standard comparison should work:
```suggestion
                    self.assertEqual(tensor.size(), expected_tensor.size())
```
This is much clearer to read and also ends up giving a much more informative error message in case of failure.

Reply:
okay Ill update it.